src/main/java/com/github/princesslana/eriscasper/rest/guild/ModifyCurrentNickRequest.java

```diff
@@ -0,0 +1,17 @@
+package com.github.princesslana.eriscasper.rest.guild;
+
+import org.immutables.value.Value;
+
+/**
+ * @see <a
+ *     href="https://discordapp.com/developers/docs/resources/guild#modify-current-user-nick-json-params">
+ *     https://discordapp.com/developers/docs/resources/guild#modify-current-user-nick-json-params</a>
+ */
+@Value.Immutable
+public interface ModifyCurrentNickRequest {
+  String getNick();
+
+  static ModifyCurrentNickRequest withNick(String nick) {
```

Comment:
`ofNick`.

It does sound a little funky, but we use of for construction of classes elsewhere. `withXxx` methods are generated by immutables and have a different use.

Reply:
gotcha